### ISSUE TITLE
No matching version found for node-pre-gyp@undefined

### ISSUE THREAD

**Issue by gonzoinc** [jameswlane/status-board#94]
Stumbled across this from the original Atlas Board which I had installed without issue. So not sure which dependency you are using that has the issue.

`sudo npm install -g status-board
npm WARN deprecated @types/chalk@2.2.0: This is a stub types definition for chalk (https://github.com/chalk/chalk). chalk provides its own type definitions, so you don't need @types/chalk installed!
npm WARN deprecated @types/moment@2.13.0: This is a stub types definition for Moment (https://github.com/moment/moment). Moment provides its own type definitions, so you don't need @types/moment installed!
npm WARN deprecated github@13.0.1: 'github' has been renamed to '@octokit/rest' (https://git.io/vNB11)
npm ERR! code ETARGET
npm ERR! notarget No matching version found for node-pre-gyp@undefined
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:`

node -v
v6.11.3

npm -v
5.6.0

react
16.2.0

Logs File

`1215 http fetch GET 304 https://registry.npmjs.org/node-pre-gyp 63ms (from cache)
1216 silly registry:manifest no matching version for node-pre-gyp@undefined in the cache. Forcing revalidation
1217 http fetch GET 304 https://registry.npmjs.org/node-pre-gyp 58ms (from cache)
1218 silly fetchPackageMetaData error for node-pre-gyp@latest No matching version found for node-pre-gyp@undefined
1219 silly saveTree 
1220 verbose type tag
1221 verbose stack node-pre-gyp: No matching version found for node-pre-gyp@undefined
1221 verbose stack     at pickManifest (/usr/local/lib/node_modules/npm/node_modules/pacote/node_modules/npm-pick-manifest/index.js:65:11)
1221 verbose stack     at fetchPackument.then.packument (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/manifest.js:52:18)
1221 verbose stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
1221 verbose stack     at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
1221 verbose stack     at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)
1221 verbose stack     at Promise._settlePromise0 (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:614:10)
1221 verbose stack     at Promise._settlePromises (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:693:18)
1221 verbose stack     at Async._drainQueue (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:133:16)
1221 verbose stack     at Async._drainQueues (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:143:10)
1221 verbose stack     at Immediate.Async.drainQueues (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:17:14)
1221 verbose stack     at runCallback (timers.js:672:20)
1221 verbose stack     at tryOnImmediate (timers.js:645:5)
1221 verbose stack     at processImmediate [as _immediateCallback] (timers.js:617:5)
1222 verbose cwd /usr/local/
1223 verbose Darwin 15.6.0
1224 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "status-board"
1225 verbose node v6.11.3
1226 verbose npm  v5.6.0
1227 error code ETARGET
1228 error notarget No matching version found for node-pre-gyp@undefined
1229 error notarget In most cases you or one of your dependencies are requesting
1229 error notarget a package version that doesn't exist.
1230 verbose exit [ 1, true ]`

**Comment by jameswlane** [comment]
@gonzoinc 
Thanks for filing a issue, this project is actively being developed. As we are continually working to improve it, feedback like this will help make it better.

First off you shouldn't install `status-board` globally, it was built to be a dependency of your project.

We are breaking out the CLI and Generator from the application into its own [application](https://github.com/jameswlane/status-board-cli) that's not yet ready to be used.

Also we are still working on documentation, so there may be some items that are incorrect.

Could you try the following:

Add Status Board as a dependency of your project:

#### Using npm
```
npm install --save-prod status-board
```
#### Using yarn
```
yarn add status-board
```

In your `start.js` file do something like:
```
const statusBoard = require('status-board').default;

statusBoard(
  {
    port: process.env.ATLASBOARD_PORT || 3000,
    install: false
  },
  function (err) {
    if (err) {
      throw err;
    }
  }
);
```

Here is a [example](https://github.com/jameswlane/status-board-example) project.

**Comment by jameswlane** [comment]
@gonzoinc After some digging I think the issue may of spawned from the included `npm-shrinkwrap.json` file. I made a few dependencies changes and rebuilt the lock files. I will be cutting a patch version to see if it resolves the issue.

**Comment by jameswlane** [comment]
When you have some time could you try and install v1.1.23 and see if your still getting it issue.

Testing on v1.1.23 I was able to install it globally and locally using:
* node@6.11.3 / npm@3.10.10
* node@6.11.3 / npm@5.6.0

So I think we may have resolved the issue.

**Comment by gonzoinc** [comment]
Thanks, That seemed to have done the trick and allowed it to install. On starting those I got errors in regards to iCal and Cheerio missing.  Once I manually installed them the board fired right up.  

First time using anything like this so it's all experimental now so not sure if I'm missing something, or a dependency is not being defined somewhere so that it gets installed. As a first time user it was also unclear that I needed Atlas board installed globally and that this is a direct module of it.  I was under the assumption this was a direct fork replacement of Atlas board seeing as it is no longer being developed. I'm moving over from an older Dashing board due to it being abandoned so glad to see development on this continues. Thanks again!
